core/lib/spree/event/adapters/event_bus.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'spree/event/event'
+require 'spree/event/listener'
+
+module Spree
+  module Event
+    module Adapters
+      # Adapter for {Spree::Event}
+      #
+      # Please, access it through {Spree::Event} module. You only need to
+      # configure an instance of it to be used as the default adapter.  E.g., in
+      # `spree.rb` initializer:
+      #
+      # @example
+      #   require "spree/event/adapters/event_bus"
+      #
+      #   Spree.config do |config|
+      #     # ...
+      #     config.events.adapter = EventBus.new
+      #     # ...
+      #   end
+      #
+      # You won't need to do that from Solidus version 4.0 as this adapter will
+      # be the default one.
+      class EventBus
+        # @api private
```

Comment:
You can apply `@api private` to the class, and YARD will mark all children of that class as private. See https://www.rubydoc.info/gems/yard/file/docs/Tags.md#api:
> Note: This tag is transitive. If it is applied on a namespace (module or class), it will immediately be applied to all children objects of that namespace unless it is redefined on the child object.

Reply:
Hmm, yeah, but the class is not private. It belongs to the public API, as you can do:

```ruby
Spree.config do |config|
  config.events.adapter = Spree::Events::Adapters::EventBus.new
ènd
```

Reply:
In fact, I just removed the `@api private` tag from the `initialize` method, as you need to call it to the bus. Please, let me know what you think :slightly_smiling_face: